packages/blade-svelte/src/components/Link/BaseLink/BaseLink.svelte

```diff
@@ -0,0 +1,304 @@
+    testID?: string;
+    htmlTitle?: string;
+    opacity?: number;
+    // Event handlers
+    onBlur?: (event: FocusEvent) => void;
+    onFocus?: (event: FocusEvent) => void;
+    onMouseLeave?: (event: MouseEvent) => void;
+    onMouseMove?: (event: MouseEvent) => void;
+    onPointerDown?: (event: PointerEvent) => void;
+    onPointerEnter?: (event: PointerEvent) => void;
+    onTouchStart?: (event: TouchEvent) => void;
+    onTouchEnd?: (event: TouchEvent) => void;
+    onMouseDown?: (event: MouseEvent) => void;
+    onMouseUp?: (event: MouseEvent) => void;
+    // Analytics attributes
+    [key: `data-analytics-${string}`]: string;
+  }
+
+  let {
+    children,
+    icon: Icon,
+    iconPosition = 'left',
+    variant = 'anchor',
+    href,
+    target,
+    rel,
+    isDisabled = false,
+    onClick,
+    color = 'primary',
+    size = 'medium',
+    accessibilityProps,
+    testID,
+    htmlTitle,
+    opacity,
+    onBlur,
+    onFocus,
+    onMouseLeave,
+    onMouseMove,
+    onPointerDown,
+    onPointerEnter,
+    onTouchStart,
+    onTouchEnd,
+    onMouseDown,
+    onMouseUp,
+    ...rest
+  }: BaseLinkProps = $props();
+
+  // Validation - check if we have either icon or children
+  if (import.meta.env.DEV && !Icon && !children) {
+    throw new Error('BaseLink: At least one of icon or text is required to render a link.');
+  }
+
+  const isButton = variant === 'button';
+  const disabled = isButton && isDisabled;
+
+  // Create interaction state using $state (must be in .svelte file)
+  let currentInteraction = $state<'default' | 'hover' | 'focus' | 'disabled'>(
+    disabled ? 'disabled' : 'default',
+  );
+
+  // Use interaction hook for managing interaction states
+  const {
+    onMouseEnter: handleMouseEnterInteraction,
+    onMouseLeave: handleMouseLeaveInteraction,
+    onFocus: handleFocusInteraction,
+    onBlur: handleBlurInteraction,
+  } = useInteraction(
+    () => currentInteraction,
+    (state: ActionStatesType) => {
+      currentInteraction = state;
+    },
+  );
+
+  // Update interaction state when disabled prop changes
+  $effect(() => {
+    if (disabled) {
+      currentInteraction = 'disabled';
+    } else if (currentInteraction === 'disabled') {
+      currentInteraction = 'default';
+    }
+  });
+
+  // Get color token based on state - reactive
+  function getColorToken(element: 'icon' | 'text'): string {
+    const state = currentInteraction;
+    const map: Record<ActionStatesType, ColorType> = {
+      default: 'normal',
+      hover: 'subtle',
+      focus: 'normal',
+      disabled: 'disabled',
+    };
+
+    const stateKey = map[state];
+    
+    if (color && color !== 'primary') {
+      if (color !== 'white') {
+        return `var(--interactive-${element}-${color}-${stateKey})`;
+      }
+      return `var(--interactive-${element}-static-white-${stateKey})`;
+    }
+    return `var(--interactive-${element}-primary-${stateKey})`;
+  }
+
+  // Get text size tokens
+  const textSizes = {
```

Comment:
you can check https://cva.style/docs if it helps in such scenarios. It helps maps variants / props to CSS styles